From 7efd72a95e00e4528ed89000ebb81d6a9b0432eb Mon Sep 17 00:00:00 2001
From: Phillip Susi <psusi@ubuntu.com>
Date: Wed, 30 Nov 2011 13:13:58 -0500
Subject: parted: make _partition_warn_busy actually a warning instead of an
 error

This function was throwing a PED_EXCEPTION_ERROR with only the
PED_EXCEPTION_CANCEL option.  Converted to a PED_EXCEPTION_WARNING with
the option to continue anyhow.

Patch-Name: 0004-parted-make-_partition_warn_busy-actually-a-warning-.patch
---
 parted/parted.c | 19 +++++++++++--------
 1 file changed, 11 insertions(+), 8 deletions(-)

diff --git a/parted/parted.c b/parted/parted.c
index 1351476..4a55173 100644
--- a/parted/parted.c
+++ b/parted/parted.c
@@ -222,14 +222,17 @@ _partition_warn_busy (PedPartition* part)
 
         if (ped_partition_is_busy (part)) {
                 path = ped_partition_get_path (part);
-                ped_exception_throw (
-                        PED_EXCEPTION_ERROR,
-                        PED_EXCEPTION_CANCEL,
-                        _("Partition %s is being used. You must unmount it "
-                          "before you modify it with Parted."),
-                        path);
-                free (path);
-                return 0;
+                if (ped_exception_throw (
+                            PED_EXCEPTION_WARNING,
+                            PED_EXCEPTION_YES_NO,
+                            _("Partition %s is being used. Are you sure you " \
+                              "want to continue?"),
+                            path) != PED_EXCEPTION_YES)
+                {
+                        free (path);
+                        return 0;
+                }
+		free (path);
         }
         return 1;
 }
